core/src/main/java/com/scalar/db/storage/cassandra/CassandraAdmin.java

```diff
@@ -98,14 +138,30 @@ public void dropTable(String namespace, String table) throws ExecutionException
 
   @Override
   public void dropNamespace(String namespace) throws ExecutionException {
-    String dropKeyspace = SchemaBuilder.dropKeyspace(quoteIfNecessary(namespace)).getQueryString();
     try {
-      clusterManager.getSession().execute(dropKeyspace);
+      dropKeyspace(namespace);
+      deleteFromKeyspacesTable(namespace);
+      dropKeyspacesTableIfEmpty();
```

Comment:
Does this mean the keyspace for metadata is removed if there is no namespace? If so, I think we can keep the namespace for metadata in this case.

Reply:
I saw similar codes in other classes and I guess you implemented this for consistency. If so, let's go with the consistent behavior for now.

Reply:
> Does this mean the keyspace for metadata is removed if there is no namespace? If so, I think we can keep the namespace for metadata in this case.
If the user has no table and namespace in the storage anymore, I think it's more kind to the user to delete any trace of ScalarDB left by removing the metadata namespace and their tables. 
What is your concern about deleting the namespace for metadata?

Reply:
Your idea sounds reasonable.

My comment above came from the following thoughts:
1. It increases the possibility of race conditions issue like this
   a. Request A is processing `dropNamespace` and just confirmed `KEYSPACES_TABLE` is empty in `dropKeyspacesTableIfEmpty`
   b. Request B is processing `createNamespace` and just created `metadataKeyspace` right after `a`
   c. Request A removes `metadataKeyspace`
   d. Request B fails to execute `insertIntoKeyspacesTable` due to removed `metadataKeyspace`
2. The checking if `KEYSPACES_TABLE` is empty is executed on every table deletion. It introduces additional duration especially in integration tests that may remove tables frequently
3. I don't think users think of an empty metadata table

`1` may not be matter since Request B can retry? And also `2` isn't critical issue. I think we can go with current behavior as my opinion above isn't strong.

Reply:
I see. Let's keep it like this.
 About 1, the Admin API is not designed to be thread-safe, so I don't think we should worry about this scenario.
 About 2, indeed, it adds an overhead when deleting table and namespace. In this case, we prioritize usability over adding a way for the user to delete the metadata namespace expressly.